container-search/src/main/java/com/yahoo/search/dispatch/Dispatcher.java

```diff
@@ -218,11 +218,16 @@ private void processResponse(Client.GetDocsumsResponseOrError responseOrError) {
         }
 
         private void fill(List<FastHit> hits, byte[] slimeBytes) {
-            Inspector summaries = new SlimeAdapter(BinaryFormat.decode(slimeBytes).get().field("docsums"));
+            Slime slime = BinaryFormat.decode(slimeBytes);
+            int maxFieldsInhit = slime.symbols();
+            Inspector summaries = new SlimeAdapter(slime.get().field("docsums"));
+            summaries.fieldCount();
```

Comment:
I think you meant to do int maxFieldInhit = summaries.fieldCount()
and remove the slime.symbols line ...

Reply:
That was the initial thought, but looking at the implementations I concluded that would only yield the value 0 as that is an array.
It might be more correct to use summaries.entry(i).getFieldCount() which should give the correct number of fields.
As symbols >= fields I used a pragmatic "large enough" approach.
@havardpe What say you ?

Reply:
If possible, I suggest performing the reserve operation inside the inner fill based on the actual number of fields in the docsum we are processing. I also suggest accounting for the 0.75 load factor somewhere to avoid a forced re-hash (possibly inside getFieldMap, since capacity is a less useful concept here than 'number of fields I can store before overhead happens').